render/render.js

```diff
@@ -851,7 +851,13 @@ module.exports = function($window) {
 		var result
 		if (typeof handler === "function") result = handler.call(ev.currentTarget, ev)
 		else if (typeof handler.handleEvent === "function") handler.handleEvent(ev)
-		if (this._ && ev.redraw !== false) (0, this._)()
+		if (this._ && ev.redraw !== false) {
+			(0, this._)()
+			if (result != null) {
+				if (typeof result.finally === "function") result.finally((0, this._))
+				else if (typeof result.then === "function") result.then((0, this._))
```

Comment:
1. Your `(0, expr)` is redundant. It's not immediately called, so it does nothing.
2. Let's do it even on error, similarly to my above `finally` to ensure errors get reported correctly. (Obviously, you can't use `async`/`await`, but it's fairly straightforward.)

Reply:
1. I was not sure about the use of this `(0, expr)` so I've tried to keep it consistent. I'm gonna call `this._` directly.
2. Not sure what you mean by "my above `finally`" but I guess I can just pass `this._` to both `resolve` and `reject` callbacks of `then`, like this:

```js
if (result != null && typeof result.then === "function") result.then(this._, this._)
```

I just need to update it with the tests as it seems to work as expected. Thank you!